stdlib/ballerina-websub/src/main/java/org/ballerinalang/net/websub/BallerinaWebSubConnectionListener.java

```diff
@@ -263,34 +241,41 @@ private BStruct createBStruct(ProgramFile programFile, String packagePath, Strin
      * @param httpCarbonMessage the message/request received
      */
     private void autoRespondToIntentVerification(HTTPCarbonMessage httpCarbonMessage) {
-        String annotatedTopic = httpCarbonMessage.getProperty(WebSubSubscriberConstants.ANNOTATED_TOPIC).toString();
+        String annotatedTopic = httpCarbonMessage.getProperty(ANNOTATED_TOPIC).toString();
         if (httpCarbonMessage.getProperty(HttpConstants.QUERY_STR) != null) {
             String queryString = (String) httpCarbonMessage.getProperty(HttpConstants.QUERY_STR);
             BMap<String, BString> params = new BMap<>();
             try {
                 HTTPCarbonMessage response = HttpUtil.createHttpCarbonMessage(false);
                 response.waitAndReleaseAllEntities();
                 URIUtil.populateQueryParamMap(queryString, params);
-                String mode = params.get(WebSubSubscriberConstants.PARAM_HUB_MODE).stringValue();
-                if ((WebSubSubscriberConstants.SUBSCRIBE.equals(mode)
-                             || WebSubSubscriberConstants.UNSUBSCRIBE.equals(mode))
-                        && annotatedTopic.equals(params.get(WebSubSubscriberConstants.PARAM_HUB_TOPIC).stringValue())) {
-                    String challenge = params.get(
-                    WebSubSubscriberConstants.PARAM_HUB_CHALLENGE).stringValue();
+                String mode = params.get(PARAM_HUB_MODE).stringValue();
+                if (!params.keySet().contains(PARAM_HUB_MODE) ||
+                        !params.keySet().contains(PARAM_HUB_TOPIC) ||
+                        !params.keySet().contains(PARAM_HUB_CHALLENGE)) {
+                    response.setProperty(HttpConstants.HTTP_STATUS_CODE, 404);
```

Comment:
Shall we use netty status codes instead of 404?
use HttpResponseStatus.NOT_FOUND.code()
Please update the other status codes as well

Reply:
Fixed all occurrences in this class with 80d51f0.